src/Log/LogService.php

```diff
@@ -0,0 +1,299 @@
+ * of Science, Research and the Arts of the State of Baden-Wuerttemberg.
+ *
+ * OPUS 4 is a complete rewrite of the original OPUS software and was developed
+ * by the Stuttgart University Library, the Library Service Center
+ * Baden-Wuerttemberg, the Cooperative Library Network Berlin-Brandenburg,
+ * the Saarland University and State Library, the Saxon State Library -
+ * Dresden State and University Library, the Bielefeld University Library and
+ * the University Library of Hamburg University of Technology with funding from
+ * the German Research Foundation and the European Regional Development Fund.
+ *
+ * LICENCE
+ * OPUS is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the Licence, or any later version.
+ * OPUS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details. You should have received a copy of the GNU General Public License
+ * along with OPUS; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ * @category opus4-common
+ * @Package Opus\Log
+ * @author      Kaustabh Barman <barman@zib.de>
+ * @copyright   Copyright (c) 2020, OPUS 4 development team
+ * @license     http://www.gnu.org/licenses/gpl.html General Public License
+ */
+
+namespace Opus\Log;
+
+/**
+ * Class to manage multiple loggers
+ *
+ * @package     Opus\Log
+ *
+ */
+class LogService
+{
+
+    /** @var Zend_Log all log objects with lognames as keys*/
+    private $loggers = [];
+
+    private $logFile;       //file pointer resource to the logfile
+
+    /** @var string Format of log output */
+    private $format;
+
+    private $config;    //Zend_Config
+
+    /** @var null path to the log file */
+    private $logPath = null;
+
+    /** @var Opus\Log\LogService Singleton instance of LogService */
+    private static $instance;
+
+    /** @var string name of the log file */
+    private $logFileName;
+
+    /**
+     * @var string name of the default log name
+     */
+    private $defaultLogName = 'default';
+
+    public static function getInstance()
+    {
+        if (null === static::$instance) {
+            static::$instance = new static();
+        }
+        return static::$instance;
+    }
+
+    protected function __construct()
+    {
+    }
+
+    /**
+     * @param $logName string name of the log
+     * @param null $logFileName string optional name of the log file
+     */
+    public function createLog($logName, $logFileName = null)
+    {
+        $this->getConfig();
+
+        $logfilePath = $this->logPath == null ? $this->setPath($this->config->workspacePath . '/log/') : $this->logPath;
+
+        if ($logFileName == null) {
```

Comment:
This is tricky. The purpose of this function is not entirely clear yet to me. I think we need it and the way it is used right ow is basically okay. I think this bit about the logFileName needs work, but I think it will have wait until the rest has been addressed, because it gets to confusing otherwise. 

Basically one part has not been addressed at all yet, configured loggers. We have some global logging configuration, basically the defaults. We could more the default format into the configuration. That might be a nice improvement, but it isn't essential right now. However we want to add configuration for specific loggers later, so we have to ways of creating loggers. We can create them dynamically at runtime using this function. Do we need additional optional parameters for that, perhaps `format` and `priority/level`? The other way is to configure the translation logger with filename, format and level in the configuration. It a logger has been configured in this way it should be enough to call `getLog('translation')` and `LogService` will create the `Zend_Log` object using the specific and default configuration. We could decide that if *translation* has not been configured, we use all the default parameters except the file name. Using the file name `translation.log` makes sense as a backup if nothing else has been provided.

For now we could add a function `getLogConfig($name)`. It it returns null, we assume there is not specific configuration and we continue using the defaults. Another option here would be to basically always return something from `getLogConfig` and merge the default and specific configuration for a logger inside that function. That might make it very clean. However I do not think the rest of the code is there yet. So I suggest adding this function with a description simply returning *null* right now. We will add the configuration feature a little later when we have figured out the rest. It gets too confusing otherwise. Too many moving parts.

Reply:
We can create loggers during runtime without needing additional parameters like `format` or `priority/level` . But we can still configure the default parameters by calling `setDefaultFormat` and `setDefaultLogLevel()` (coming up on next commit). Also, calling `getlog('translation') will create a Zend_Log object with the default configurations. The file name will also be `translation.log` by default.